### PR TITLE
Fat 1715/c350540

### PR DESCRIPTION
**Pull request by kimkate303**
https://foliotest.testrail.io/index.php?/cases/view/350540

### CODE REVIEW

File: cypress/integration/requests/patron-initiated/request-item-recall.spec.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+  };
+  const patronGroup = {};
+  const userData0 = { ...DefaultUser.defaultApiPatron };
+  const userData1 = { ...DefaultUser.defaultApiPatron };
+  const ITEM_BARCODE = generateItemBarcode();
+  userData1.username = `autotest_username_${getRandomPostfix()}`;
+  userData1.barcode = `1234456_${getRandomPostfix()}`;
+  requestPolicyWithRecall.requestTypes.push('Recall', 'Page');
+
+  console.log(requestPolicyWithRecall);
```
Comment (IhorBohdan) [inline cypress/integration/requests/patron-initiated/request-item-recall.spec.js:36]: Does need here console.log?

Comment (kimkate303) [reply cypress/integration/requests/patron-initiated/request-item-recall.spec.js:36]: omg) deleted, thanks)

File: cypress/integration/requests/patron-initiated/request-item-recall.spec.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        Users.createViaApi({
+          patronGroup: res,
+          ...userData1
+        }).then((createdUser) => {
+          userData1.id = createdUser.id;
+        });
+      });
+
+    // Create item with remembered instance type
+    ServicePoints.getViaApi({ limit: 1, query: 'pickupLocation=="true"' })
```
Comment (MaksimTerekhov) [inline cypress/integration/requests/patron-initiated/request-item-recall.spec.js:66]: try to use InventoryInstances.createInstanceViaApi
Or you can create your own creating function

File: cypress/integration/requests/patron-initiated/request-item-recall.spec.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      Requests.getRequestIdViaApi({ limit:1, query: `item.barcode="${ITEM_BARCODE}"` })
+        .then(requestId => {
+          Requests.deleteRequestApi(requestId);
+          Users.deleteViaApi(userData0.id);
+          Users.deleteViaApi(userData1.id);
+          PatronGroups.deleteViaApi(patronGroup.id);
+        });
+    });
+
+    cy.getInstance({ limit: 1, expandAll: true, query: `"items.barcode"=="${ITEM_BARCODE}"` })
```
Comment (MaksimTerekhov) [inline cypress/integration/requests/patron-initiated/request-item-recall.spec.js:134]: InventoryInstances.deleteInstanceAndHoldingRecordAndAllItemsViaApi

File: cypress/integration/requests/patron-initiated/request-item-recall.spec.js (hunk 2 above)
Comment (kimkate303) [reply cypress/integration/requests/patron-initiated/request-item-recall.spec.js:78]: done)

File: cypress/integration/requests/patron-initiated/request-item-recall.spec.js (hunk 3 above)
Comment (kimkate303) [reply cypress/integration/requests/patron-initiated/request-item-recall.spec.js]: done)

**Review by MaksimTerekhov — APPROVED** [review]

**Review by IhorBohdan — APPROVED** [review]

File: cypress/integration/requests/patron-initiated/request-item-recall.spec.js [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+import Users from '../../../support/fragments/users/users';
+import DefaultUser from '../../../support/fragments/users/userDefaultObjects/defaultUser';
+import ServicePoints from '../../../support/fragments/settings/tenant/servicePoints/servicePoints';
+import UserEdit from '../../../support/fragments/users/userEdit';
+import CirculationRules from '../../../support/fragments/circulation/circulation-rules';
+import Checkout from '../../../support/fragments/checkout/checkout';
+import CheckInActions from '../../../support/fragments/check-in-actions/checkInActions';
+import Requests from '../../../support/fragments/requests/requests';
+import NewRequest from '../../../support/fragments/requests/newRequest';
+import inventoryInstances from '../../../support/fragments/inventory/inventoryInstances';
```
Comment (penfriend) [inline cypress/integration/requests/patron-initiated/request-item-recall.spec.js:18]: please import a fragment starting from the UpperCase letter

File: cypress/integration/requests/patron-initiated/request-item-recall.spec.js [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+      });
+
+
+    // Create a circulation rule with an associated request policy that allows recalls and pages
+    cy.getCirculationRules().then((res) => {
+      testData.baseRules = res.rulesAsText;
+      testData.ruleProps = CirculationRules.getRuleProps(res.rulesAsText);
+      testData.ruleProps.r = testData.requestPolicyId;
+
+      CirculationRules.addRuleApi(res.rulesAsText, testData.ruleProps, 'g ', testData.patronGroup.id).then(() => {
```
Comment (penfriend) [inline cypress/integration/requests/patron-initiated/request-item-recall.spec.js:92]: I have not founded the precondition of the test case related to the special loan police:
Item(s) that are checked out using a loan policy that uses rolling loan periods

Can you add please a loan policy that uses rolling loan periods?

**Comment by kimkate303** [comment]
> Please create the loan policy that uses rolling loan periods, add it to the circulation rule and remove it after the test execution

The two default loan policies that are used in circ rules are both use rolling periods

**Comment by penfriend** [comment]
> > Please create the loan policy that uses rolling loan periods, add it to the circulation rule and remove it after the test execution
> 
> The two default loan policies that are used in circ rules are both use rolling periods

Our tests shouldn't depend on the default policy because different environments can have different default policies.

**Comment by penfriend** [comment]
I have some doubts that we need to test here using UI. There is a diagram that shows that we need to call the backend module mod-patron directly. We need to clarify this asking the PO.
https://wiki.folio.org/display/FOLIOtips/Edge+APIs?preview=/25723203/25723207/EdgePatron.png#EdgeAPIs-Diagram.1

**Comment by penfriend** [comment]
Please add the ability to have an option of TLR request and select this option during the creation of the TLR request

**Pull request closed without merging**